### PR TITLE
feat: Graceful Shutdowns

### PR DESCRIPTION
**Pull request by opaolini**
# Description

This PR adds a `createDefaultServer` function to be used across all HTTP runners which prepares a `http.Server` with:
- Header timeouts `server.keepAliveTimeout` &  `server.headersTimeout`
- Healthcheck service exposed at `/healthz`
- Graceful shutdown through [server.close](https://nodejs.org/api/http.html#http_server_close_callback)
- Adds common middleware: `requestLogger`, `cors`, `bodyParser`

# Testing Instructions
This PR has been tested on staging, however we can still unit test the graceful shutdown:
- [ ] Add unit test for graceful shutdowns

# Checklist

-   [x] Update [documentation](https://github.com/0xProject/website/blob/development/mdx/api/index.mdx) as needed. **Website Documentation PR:**
-   [x] Prefix PR title with `[WIP]` if necessary.
-   [ ] Add tests to cover changes as needed.

### CODE REVIEW

**Comment by opaolini** [comment]
deploy staging

**Comment by opaolini** [comment]
deploy staging

File: src/runners/utils.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        healthcheckService.setHealth(false);
+        server.close(async err => {
+            if (!server.listening) {
+                process.exit(0);
+            }
+            if (dependencies.connection) {
+                await dependencies.connection.close();
+            }
+            if (dependencies.meshClient) {
+                dependencies.meshClient.destroy();
```
Comment (xianny) [inline src/runners/utils.ts:53]: just wanna confirm this is **not** async, right?

Comment (opaolini) [reply src/runners/utils.ts:53]: Correct, it is not async :)

File: src/config.ts [hunk 2]
```diff
@@ -50,6 +50,11 @@ export const HTTP_PORT = _.isEmpty(process.env.HTTP_PORT)
     ? 3000
     : assertEnvVarType('HTTP_PORT', process.env.HTTP_PORT, EnvVarType.Port);
 
+// Network port for the healthcheck service at /healthz, if no provided, it uses the HTTP_PORT value.
```
Comment (dekz) [inline src/config.ts:53]:
```suggestion
// Network port for the healthcheck service at /healthz, if not provided, it uses the HTTP_PORT value.
```

**Comment by opaolini** [comment]
deploy staging

**Pull request merged**